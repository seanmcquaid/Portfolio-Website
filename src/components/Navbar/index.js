import React, { useState } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { IoMdMusicalNotes } from 'react-icons/io';
import { GiHamburgerMenu } from 'react-icons/gi';
import DesktopNav from './DesktopNav';
import MobileNav from './MobileNav';

const Navbar = () => {
  const [isMobileMenuDisplayed, setIsMobileMenuDisplayed] = useState(false);

  const hamburgerMenuOnClick = () => {
    setIsMobileMenuDisplayed((prevState) => !prevState);
  };

  const mobileMenuItemOnClick = () => {
    setIsMobileMenuDisplayed(false);
  };

  return (
    <NavbarContainer data-testid="navbar" id="navbar">
      <LinkContainer
        to="/"
        onClick={mobileMenuItemOnClick}
        data-testid="homeButton"
        id="homeButton"
        title="Home Page Link"
      >
        <IoMdMusicalNotes />
      </LinkContainer>
      <DesktopNav />
      <MobileNav
        isMobileMenuDisplayed={isMobileMenuDisplayed}
        mobileMenuItemOnClick={mobileMenuItemOnClick}
      />
      <HamburgerIcon
        onClick={hamburgerMenuOnClick}
        data-testid="mobileMenuToggleButton"
        id="mobileMenuToggleButton"
        title="Mobile Nav Toggle"
      />
    </NavbarContainer>
  );
};

const NavbarContainer = styled.nav`
  background-color: rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  top: 0;
  left: 0;
  z-index: 1;
  width: 100px;
  height: 100%;
  @media (max-width: 975px) {
    flex-direction: row;
    width: 100%;
    height: 80px;
    position: fixed;
  }
  @media (max-height: 620px) {
    flex-direction: row;
    width: 100%;
    height: 80px;
    position: fixed;
  }
`;

const LinkContainer = styled(Link)`
  padding: 1rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 1.5rem;
  color: #ffffffb6;
  text-decoration: none;
`;

const HamburgerIcon = styled(GiHamburgerMenu)`
  font-size: 1.5rem;
  padding: 1rem;
  &:hover {
    cursor: pointer;
  }
  @media (min-width: 975px) {
    display: none;
  }
  @media (max-height: 620px) {
    display: block;
  }
`;

export default Navbar;
