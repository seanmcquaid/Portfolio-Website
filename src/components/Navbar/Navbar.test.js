import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Navbar from './Navbar';
import '@testing-library/jest-dom/extend-expect';

describe('<Navbar/>', () => {
  it('Matches snapshot', () => {
    const history = createMemoryHistory();
    const navbar = render(
      <Router history={history}>
        <Navbar />
      </Router>
    );
    expect(navbar).toMatchSnapshot();
  });

  it('Hamburger menu toggle works', () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router history={history}>
        <Navbar />
      </Router>
    );

    act(() => {
      window.innerWidth = 320;
      window.innerHeight = 568;

      fireEvent(window, new Event('resize'));
    });

    fireEvent.click(getByTestId('mobileMenuToggleButton'));

    expect(getByTestId('mobileAboutLink')).toBeVisible();
  });

  it('Mobile menu toggle works', () => {
    const history = createMemoryHistory();

    const { getByTestId } = render(
      <Router history={history}>
        <Navbar />
      </Router>
    );

    act(() => {
      window.innerWidth = 320;
      window.innerHeight = 568;

      fireEvent(window, new Event('resize'));
    });

    fireEvent.click(getByTestId('mobileMenuToggleButton'));

    fireEvent.click(getByTestId('homeButton'));

    expect(getByTestId('mobileAboutLink')).not.toBeVisible();
  });
});
