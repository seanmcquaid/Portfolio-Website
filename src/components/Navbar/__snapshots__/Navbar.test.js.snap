// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar/> Matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm VfmIA"
        data-testid="navbar"
        id="navbar"
      >
        <a
          class="sc-AxirZ esVepc"
          data-testid="homeButton"
          href="#/"
          id="homeButton"
        >
          <div
            class="sc-AxgMl mcXLr"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M160 64v257.6c-8.2-2.7-17.2-4.1-26.6-4.1-38.3 0-69.4 27.1-69.4 65.4 0 38.3 31.1 65.1 69.4 65.1 38.3 0 69.6-28.2 69.6-69.1V200h202v121.6c-8.2-2.7-17.2-4.1-26.6-4.1-38.3 0-69.4 27.1-69.4 65.4 0 38.3 31.1 65.1 69.4 65.1 38.3 0 69.6-28.2 69.6-69.1V64H160zm245 96H203v-53h202v53z"
              />
            </svg>
          </div>
          <span
            class="sc-AxhUy gLXoPp"
          >
            Home
          </span>
        </a>
        <nav
          class="sc-AxiKw bPcyZw"
          data-testid="desktopNavMenu"
          id="desktopNavMenu"
        >
          <a
            class="sc-AxhCb kQBoVz"
            data-testid="desktopAboutLink"
            href="#/about"
            id="desktopAboutLink"
          >
            <div
              class="sc-AxgMl mcXLr"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 256c52.805 0 96-43.201 96-96s-43.195-96-96-96-96 43.201-96 96 43.195 96 96 96zm0 48c-63.598 0-192 32.402-192 96v48h384v-48c0-63.598-128.402-96-192-96z"
                />
              </svg>
            </div>
            <span
              class="sc-AxhUy gLXoPp"
            >
              About
            </span>
          </a>
          <a
            class="sc-AxhCb kQBoVz"
            data-testid="desktopSkillsLink"
            href="#/skills"
            id="desktopSkillsLink"
          >
            <div
              class="sc-AxgMl mcXLr"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
                />
              </svg>
            </div>
            <span
              class="sc-AxhUy gLXoPp"
            >
              Skills
            </span>
          </a>
          <a
            class="sc-AxhCb kQBoVz"
            data-testid="desktopProjectsLink"
            href="#/projects"
            id="desktopProjectsLink"
          >
            <div
              class="sc-AxgMl mcXLr"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M449.2 208H423v-32l-14.4-48H383V96l-15-48H144l-15 48v32h-25.6L89 176v32H62.8L48 256v165.3c0 23.5 35.2 42.7 58.7 42.7h314.7c21.8 0 42.7-19.7 42.7-41V256l-14.9-48zM176 96h160v32H176V96zm-41 80h242v32H135v-32zm282 112h-82.6c-7.4 36.5-39.7 64-78.4 64s-71-27.5-78.4-64H95v-32h322v32z"
                />
              </svg>
            </div>
            <span
              class="sc-AxhUy gLXoPp"
            >
              Projects
            </span>
          </a>
          <a
            class="sc-AxhCb kQBoVz"
            data-testid="desktopContactLink"
            href="#/contact"
            id="desktopContactLink"
          >
            <div
              class="sc-AxgMl mcXLr"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M437.332 80H74.668C51.199 80 32 99.198 32 122.667v266.666C32 412.802 51.199 432 74.668 432h362.664C460.801 432 480 412.802 480 389.333V122.667C480 99.198 460.801 80 437.332 80zM432 170.667L256 288 80 170.667V128l176 117.333L432 128v42.667z"
                />
              </svg>
            </div>
            <span
              class="sc-AxhUy gLXoPp"
            >
              Contact
            </span>
          </a>
        </nav>
        <svg
          class="sc-fzplWN diOAb"
          data-testid="mobileMenuToggleButton"
          fill="currentColor"
          height="1em"
          id="mobileMenuToggleButton"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
          />
        </svg>
        <div
          class="sc-fzpans hGqZeR"
          data-testid="mobileNavMenu"
          id="mobileMenuToggleButton"
        >
          <nav
            class="sc-fznyAO jefWsP"
          >
            <a
              class="sc-fznKkj dGhEQm"
              data-testid="mobileAboutLink"
              href="#/about"
              id="mobileAboutLink"
            >
              <span
                class="sc-fznZeY eWcBjj"
              >
                About
              </span>
            </a>
            <a
              class="sc-fznKkj dGhEQm"
              data-testid="mobileSkillsLink"
              href="#/skills"
              id="mobileSkillsLink"
            >
              <span
                class="sc-fznZeY eWcBjj"
              >
                Skills
              </span>
            </a>
            <a
              class="sc-fznKkj dGhEQm"
              data-testid="mobileProjectsLink"
              href="#/projects"
              id="mobileProjectsLink"
            >
              <span
                class="sc-fznZeY eWcBjj"
              >
                Projects
              </span>
            </a>
            <a
              class="sc-fznKkj dGhEQm"
              data-testid="mobileContactLink"
              href="#/contact"
              id="mobileContactLink"
            >
              <span
                class="sc-fznZeY eWcBjj"
              >
                Contact
              </span>
            </a>
          </nav>
        </div>
        <ul
          class="sc-AxheI jMIuVM"
          data-testid="desktopContactMenu"
          id="desktopContactMenu"
        >
          <li
            class="sc-AxmLO bsqIBN"
          >
            <a
              class="sc-fzoLsD eVaKKO"
              data-testid="desktopLinkedInLink"
              href="https://www.linkedin.com/in/sean-mcquaid-292b3588/"
              id="desktopLinkedInLink"
              target="blank"
            >
              <div
                class="sc-fzozJi EoHts"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M417.2 64H96.8C79.3 64 64 76.6 64 93.9V415c0 17.4 15.3 32.9 32.8 32.9h320.3c17.6 0 30.8-15.6 30.8-32.9V93.9C448 76.6 434.7 64 417.2 64zM183 384h-55V213h55v171zm-25.6-197h-.4c-17.6 0-29-13.1-29-29.5 0-16.7 11.7-29.5 29.7-29.5s29 12.7 29.4 29.5c0 16.4-11.4 29.5-29.7 29.5zM384 384h-55v-93.5c0-22.4-8-37.7-27.9-37.7-15.2 0-24.2 10.3-28.2 20.3-1.5 3.6-1.9 8.5-1.9 13.5V384h-55V213h55v23.8c8-11.4 20.5-27.8 49.6-27.8 36.1 0 63.4 23.8 63.4 75.1V384z"
                  />
                </svg>
              </div>
              <span
                class="sc-Axmtr kcFsFG"
              >
                LinkedIn
              </span>
            </a>
          </li>
          <li
            class="sc-AxmLO bsqIBN"
          >
            <a
              class="sc-fzoLsD eVaKKO"
              data-testid="desktopGitHubLink"
              href="https://github.com/seanmcquaid"
              id="desktopGitHubLink"
              target="blank"
            >
              <div
                class="sc-fzozJi EoHts"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9 1.4.3 2.6.4 3.8.4 8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1-8.4 1.9-15.9 2.7-22.6 2.7-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1 10.5 0 20-3.4 25.6-6 2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8 0 0 1.6-.5 5-.5 8.1 0 26.4 3.1 56.6 24.1 17.9-5.1 37-7.6 56.1-7.7 19 .1 38.2 2.6 56.1 7.7 30.2-21 48.5-24.1 56.6-24.1 3.4 0 5 .5 5 .5 12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5 1.2 0 2.6-.1 4-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
                  />
                </svg>
              </div>
              <span
                class="sc-Axmtr kcFsFG"
              >
                GitHub
              </span>
            </a>
          </li>
          <li
            class="sc-AxmLO bsqIBN"
          >
            <a
              class="sc-fzoLsD eVaKKO"
              data-testid="desktopResumeLink"
              href="Sean McQuaid - Resume.pdf"
              id="desktopResumeLink"
              target="blank"
            >
              <div
                class="sc-fzozJi EoHts"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M124 80v322c0 7.7-6.3 14-14 14s-14-6.3-14-14V112H80c-17.7 0-32 14.3-32 32v288c0 17.7 14.3 32 32 32h353.1c17 0 30.9-13.8 30.9-30.9V80c0-17.7-14.3-32-32-32l-278 2c-17.7 0-30 12.3-30 30zm66 32h84c7.7 0 14 6.3 14 14s-6.3 14-14 14h-84c-7.7 0-14-6.3-14-14s6.3-14 14-14zm0 160h148c7.7 0 14 6.3 14 14s-6.3 14-14 14H190c-7.7 0-14-6.3-14-14s6.3-14 14-14zm196 108H190c-7.7 0-14-6.3-14-14s6.3-14 14-14h196c7.7 0 14 6.3 14 14s-6.3 14-14 14zm0-160H190c-7.7 0-14-6.3-14-14s6.3-14 14-14h196c7.7 0 14 6.3 14 14s-6.3 14-14 14z"
                  />
                </svg>
              </div>
              <span
                class="sc-Axmtr kcFsFG"
              >
                Resume
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm VfmIA"
      data-testid="navbar"
      id="navbar"
    >
      <a
        class="sc-AxirZ esVepc"
        data-testid="homeButton"
        href="#/"
        id="homeButton"
      >
        <div
          class="sc-AxgMl mcXLr"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M160 64v257.6c-8.2-2.7-17.2-4.1-26.6-4.1-38.3 0-69.4 27.1-69.4 65.4 0 38.3 31.1 65.1 69.4 65.1 38.3 0 69.6-28.2 69.6-69.1V200h202v121.6c-8.2-2.7-17.2-4.1-26.6-4.1-38.3 0-69.4 27.1-69.4 65.4 0 38.3 31.1 65.1 69.4 65.1 38.3 0 69.6-28.2 69.6-69.1V64H160zm245 96H203v-53h202v53z"
            />
          </svg>
        </div>
        <span
          class="sc-AxhUy gLXoPp"
        >
          Home
        </span>
      </a>
      <nav
        class="sc-AxiKw bPcyZw"
        data-testid="desktopNavMenu"
        id="desktopNavMenu"
      >
        <a
          class="sc-AxhCb kQBoVz"
          data-testid="desktopAboutLink"
          href="#/about"
          id="desktopAboutLink"
        >
          <div
            class="sc-AxgMl mcXLr"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 256c52.805 0 96-43.201 96-96s-43.195-96-96-96-96 43.201-96 96 43.195 96 96 96zm0 48c-63.598 0-192 32.402-192 96v48h384v-48c0-63.598-128.402-96-192-96z"
              />
            </svg>
          </div>
          <span
            class="sc-AxhUy gLXoPp"
          >
            About
          </span>
        </a>
        <a
          class="sc-AxhCb kQBoVz"
          data-testid="desktopSkillsLink"
          href="#/skills"
          id="desktopSkillsLink"
        >
          <div
            class="sc-AxgMl mcXLr"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M413.967 276.8c1.06-6.235 1.06-13.518 1.06-20.8s-1.06-13.518-1.06-20.8l44.667-34.318c4.26-3.118 5.319-8.317 2.13-13.518L418.215 115.6c-2.129-4.164-8.507-6.235-12.767-4.164l-53.186 20.801c-10.638-8.318-23.394-15.601-36.16-20.801l-7.448-55.117c-1.06-4.154-5.319-8.318-10.638-8.318h-85.098c-5.318 0-9.577 4.164-10.637 8.318l-8.508 55.117c-12.767 5.2-24.464 12.482-36.171 20.801l-53.186-20.801c-5.319-2.071-10.638 0-12.767 4.164L49.1 187.365c-2.119 4.153-1.061 10.399 2.129 13.518L96.97 235.2c0 7.282-1.06 13.518-1.06 20.8s1.06 13.518 1.06 20.8l-44.668 34.318c-4.26 3.118-5.318 8.317-2.13 13.518L92.721 396.4c2.13 4.164 8.508 6.235 12.767 4.164l53.187-20.801c10.637 8.318 23.394 15.601 36.16 20.801l8.508 55.117c1.069 5.2 5.318 8.318 10.637 8.318h85.098c5.319 0 9.578-4.164 10.638-8.318l8.518-55.117c12.757-5.2 24.464-12.482 36.16-20.801l53.187 20.801c5.318 2.071 10.637 0 12.767-4.164l42.549-71.765c2.129-4.153 1.06-10.399-2.13-13.518l-46.8-34.317zm-158.499 52c-41.489 0-74.46-32.235-74.46-72.8s32.971-72.8 74.46-72.8 74.461 32.235 74.461 72.8-32.972 72.8-74.461 72.8z"
              />
            </svg>
          </div>
          <span
            class="sc-AxhUy gLXoPp"
          >
            Skills
          </span>
        </a>
        <a
          class="sc-AxhCb kQBoVz"
          data-testid="desktopProjectsLink"
          href="#/projects"
          id="desktopProjectsLink"
        >
          <div
            class="sc-AxgMl mcXLr"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M449.2 208H423v-32l-14.4-48H383V96l-15-48H144l-15 48v32h-25.6L89 176v32H62.8L48 256v165.3c0 23.5 35.2 42.7 58.7 42.7h314.7c21.8 0 42.7-19.7 42.7-41V256l-14.9-48zM176 96h160v32H176V96zm-41 80h242v32H135v-32zm282 112h-82.6c-7.4 36.5-39.7 64-78.4 64s-71-27.5-78.4-64H95v-32h322v32z"
              />
            </svg>
          </div>
          <span
            class="sc-AxhUy gLXoPp"
          >
            Projects
          </span>
        </a>
        <a
          class="sc-AxhCb kQBoVz"
          data-testid="desktopContactLink"
          href="#/contact"
          id="desktopContactLink"
        >
          <div
            class="sc-AxgMl mcXLr"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M437.332 80H74.668C51.199 80 32 99.198 32 122.667v266.666C32 412.802 51.199 432 74.668 432h362.664C460.801 432 480 412.802 480 389.333V122.667C480 99.198 460.801 80 437.332 80zM432 170.667L256 288 80 170.667V128l176 117.333L432 128v42.667z"
              />
            </svg>
          </div>
          <span
            class="sc-AxhUy gLXoPp"
          >
            Contact
          </span>
        </a>
      </nav>
      <svg
        class="sc-fzplWN diOAb"
        data-testid="mobileMenuToggleButton"
        fill="currentColor"
        height="1em"
        id="mobileMenuToggleButton"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
        />
      </svg>
      <div
        class="sc-fzpans hGqZeR"
        data-testid="mobileNavMenu"
        id="mobileMenuToggleButton"
      >
        <nav
          class="sc-fznyAO jefWsP"
        >
          <a
            class="sc-fznKkj dGhEQm"
            data-testid="mobileAboutLink"
            href="#/about"
            id="mobileAboutLink"
          >
            <span
              class="sc-fznZeY eWcBjj"
            >
              About
            </span>
          </a>
          <a
            class="sc-fznKkj dGhEQm"
            data-testid="mobileSkillsLink"
            href="#/skills"
            id="mobileSkillsLink"
          >
            <span
              class="sc-fznZeY eWcBjj"
            >
              Skills
            </span>
          </a>
          <a
            class="sc-fznKkj dGhEQm"
            data-testid="mobileProjectsLink"
            href="#/projects"
            id="mobileProjectsLink"
          >
            <span
              class="sc-fznZeY eWcBjj"
            >
              Projects
            </span>
          </a>
          <a
            class="sc-fznKkj dGhEQm"
            data-testid="mobileContactLink"
            href="#/contact"
            id="mobileContactLink"
          >
            <span
              class="sc-fznZeY eWcBjj"
            >
              Contact
            </span>
          </a>
        </nav>
      </div>
      <ul
        class="sc-AxheI jMIuVM"
        data-testid="desktopContactMenu"
        id="desktopContactMenu"
      >
        <li
          class="sc-AxmLO bsqIBN"
        >
          <a
            class="sc-fzoLsD eVaKKO"
            data-testid="desktopLinkedInLink"
            href="https://www.linkedin.com/in/sean-mcquaid-292b3588/"
            id="desktopLinkedInLink"
            target="blank"
          >
            <div
              class="sc-fzozJi EoHts"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M417.2 64H96.8C79.3 64 64 76.6 64 93.9V415c0 17.4 15.3 32.9 32.8 32.9h320.3c17.6 0 30.8-15.6 30.8-32.9V93.9C448 76.6 434.7 64 417.2 64zM183 384h-55V213h55v171zm-25.6-197h-.4c-17.6 0-29-13.1-29-29.5 0-16.7 11.7-29.5 29.7-29.5s29 12.7 29.4 29.5c0 16.4-11.4 29.5-29.7 29.5zM384 384h-55v-93.5c0-22.4-8-37.7-27.9-37.7-15.2 0-24.2 10.3-28.2 20.3-1.5 3.6-1.9 8.5-1.9 13.5V384h-55V213h55v23.8c8-11.4 20.5-27.8 49.6-27.8 36.1 0 63.4 23.8 63.4 75.1V384z"
                />
              </svg>
            </div>
            <span
              class="sc-Axmtr kcFsFG"
            >
              LinkedIn
            </span>
          </a>
        </li>
        <li
          class="sc-AxmLO bsqIBN"
        >
          <a
            class="sc-fzoLsD eVaKKO"
            data-testid="desktopGitHubLink"
            href="https://github.com/seanmcquaid"
            id="desktopGitHubLink"
            target="blank"
          >
            <div
              class="sc-fzozJi EoHts"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9 1.4.3 2.6.4 3.8.4 8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1-8.4 1.9-15.9 2.7-22.6 2.7-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1 10.5 0 20-3.4 25.6-6 2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8 0 0 1.6-.5 5-.5 8.1 0 26.4 3.1 56.6 24.1 17.9-5.1 37-7.6 56.1-7.7 19 .1 38.2 2.6 56.1 7.7 30.2-21 48.5-24.1 56.6-24.1 3.4 0 5 .5 5 .5 12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5 1.2 0 2.6-.1 4-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
                />
              </svg>
            </div>
            <span
              class="sc-Axmtr kcFsFG"
            >
              GitHub
            </span>
          </a>
        </li>
        <li
          class="sc-AxmLO bsqIBN"
        >
          <a
            class="sc-fzoLsD eVaKKO"
            data-testid="desktopResumeLink"
            href="Sean McQuaid - Resume.pdf"
            id="desktopResumeLink"
            target="blank"
          >
            <div
              class="sc-fzozJi EoHts"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M124 80v322c0 7.7-6.3 14-14 14s-14-6.3-14-14V112H80c-17.7 0-32 14.3-32 32v288c0 17.7 14.3 32 32 32h353.1c17 0 30.9-13.8 30.9-30.9V80c0-17.7-14.3-32-32-32l-278 2c-17.7 0-30 12.3-30 30zm66 32h84c7.7 0 14 6.3 14 14s-6.3 14-14 14h-84c-7.7 0-14-6.3-14-14s6.3-14 14-14zm0 160h148c7.7 0 14 6.3 14 14s-6.3 14-14 14H190c-7.7 0-14-6.3-14-14s6.3-14 14-14zm196 108H190c-7.7 0-14-6.3-14-14s6.3-14 14-14h196c7.7 0 14 6.3 14 14s-6.3 14-14 14zm0-160H190c-7.7 0-14-6.3-14-14s6.3-14 14-14h196c7.7 0 14 6.3 14 14s-6.3 14-14 14z"
                />
              </svg>
            </div>
            <span
              class="sc-Axmtr kcFsFG"
            >
              Resume
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
